Features/Context/PlatformUI.php

```diff
@@ -59,28 +61,31 @@ class PlatformUI extends Context
     private $newPathsMap = array();
 
     /**
-     * Waits for Javascript to finish by checking the loading tags of the page.
+     * Behat spin functions
+     * causes waiting while a a certain function does not return true
+     * waits while an element is not present.
      */
-    protected function waitForLoadings()
+    public function spin($lambda, $wait = 30)
     {
-        $page = $this->getSession()->getPage();
-        $loadingClasses = array(
-            '.yui3-app-transitioning',
-            '.is-app-loading',
-            '.is-app-transitioning',
-            // content tree
-            '.ez-view-treeactionview.is-expanded .ez-view-treeview:not(.is-tree-loaded)',
-            '.ez-view-universaldiscoverybrowseview .ez-view-treeview:not(.is-tree-loaded)',
-            '.is-tree-node-loading',
-            // contenttype menu
-            '.ez-view-createcontentactionview.is-expanded:not(.is-contenttypeselector-loaded)',
-        );
-        $loadingSelector = implode(',', $loadingClasses);
-        while ($page->find('css', $loadingSelector) != null) {
-            usleep(100 * 1000); // 100ms
+        for ($i = 0; $i < $wait; ++$i) {
```

Comment:
@dpobel the reason we do not use the `$session::wait` it's because AFAIK it does not work with the exceptions, but the main focus of this PR is the functionality if it works then I will improve the code overall, maybe the `$session::wait` might be used I will investigate further after we check how it behaves on travis.